open-company/open-company-web
# Comments polish

Implement [this](https://projects.invisionapp.com/share/WSERRWJED) new design for comments. Feedback and bug fixes from [this](https://docs.google.com/document/d/1X3j869Y8n94zfqaGVhrWf8VOlr3qEa_4a1fxtIjH6fs/edit#heading=h.r6mvzpsya9kl) GDoc.

Make sure you run [`interaction/new-reactions`](https://github.com/open-company/open-company-interaction/tree/new-reactions).

To test:
- turn _Show scroll bars_ option of the General page (Mac only) to _Always_
- [x] do you NOT see the horizontal scroll bar in the right column of the activity modal with 0,1 or n comments? Good
- [x] do you NOT see the horizontal scroll bar in the add comment box when its' expanded? Good
- [x] do you see the vertical scroll bar in the right column of the modal only when you have scrolling comments? Good
- [x] do you see the vertical scroll bar in the add comment box only when you insert enough text? Good
- [x] with 0 comments: do you see the empty placeholder? is it horizontally centred? Good
- [x] with 1 or 2 (no scrolling comments) do you see them well placed? Good
- [x] enter enough comments to have scrolling comments. Does it scroll as it should? Do the comments start from the bottom (last added) and you can scroll up to the top? Good
- [x] scroll to the top of the comments list, focus the add comment field, does the comments scroll to the last so you know where your comments will be added? Good
- [x] blur the add comment w/o adding text, does it collapse like it was initially? Good
- [x] does it expand and scroll the comments back to the bottom if you focus again? Good
- [x] enter some text and blur the add comment field, does the add comments remain expanded? Good
- create 3 users: A, B and C all with at least author access
- add a comment with A
- [x] can you NOT see the react button below your comment? is the comment backgorund yellow? Good
- [x] check the comment w/ user B and C, can you react? is the comment background gray? Good
- react with B to the comment
- [x] does A now see the react button with 1 besides? Good
- [x] can A NOT react to his own comment? is the thumb up gray and it stays like it if you hover?
 Good
- [x] can C see the react button with 1 beside? is it gray? does it turn yellow if you hover it? Good
- [x] react with C, can B and A see the new number? Good
- add another comment with A that is a simple single emoji like 😄
- [x] does it NOT have a background (yellow or gray)? Good
- [x] do B and C see the new comment w/o background? Good
- [x] do you NOT see the react button below the single emoji comment with any of the users? Good
- [x] repeat the steps above picking random single emojis like flags 🇮🇲  and family 👨‍👩‍👦‍👦 and any other weird emoji you can think of

WIP missing test steps:
- no comment added tooltip
- edit comment
- delete comment
- emoji picker button in collapsed and expanded add comment
